hoot-core/src/main/cpp/hoot/core/io/OsmGeoJsonReader.cpp

```diff
@@ -273,7 +254,17 @@ void OsmGeoJsonReader::_parseGeoJsonNode(const string& id, const pt::ptree& prop
   //  Get info we need to construct our node
   long node_id = -1;
   if (_useDataSourceIds)
-    node_id = boost::lexical_cast<long>(id);
+  {
+    //  Try to cast the ID to a long, if unsuccessful create a new ID
+    try
+    {
+      node_id = boost::lexical_cast<long>(id);
+    }
+    catch (const boost::bad_lexical_cast&)
+    {
+      node_id = _map->createNextNodeId();
+    }
+  }
```

Comment:
Found a few examples of WFS data where the ID isn't necessarily a `long` integer.  If the `lexical_cast` fails because it isn't an integer, the rest of the code will make an ID tag with the string ID and create a new ID on the OSM object.  This is repeated for nodes, ways, and relations.